src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -1079,5 +1079,98 @@ BOOST_AUTO_TEST_CASE(SplitByAnyOfTfIdfEncodingSerialization)
   CheckMatrices(output, xmlOutput, textOutput, binaryOutput);
 }
 
+// Following Test will be changed
+
+/**
+ * Test the Tf-Idf Encoding using rawcount type and smoothidf as true,
+ * which is the deafult values with ngram range 4-5.
+ */
+BOOST_AUTO_TEST_CASE(NgramRawCountSmoothIdfEncodingTest)
+{
+  using DictionaryType = StringEncodingDictionary<boost::string_view>;
+
+//! Common input for some tests.
+static vector<string> stringInput = {
+    "this is the first document.",
+    "this document is the second document.",
+    "and this is the third one.",
+    "is this the first document."
+};
+
+  arma::mat output;
+  TfIdfEncoding<SplitByAnyOf::TokenType> encoder;
+  SplitByAnyOf tokenizer(" ,.");
+  encoder.SetmaxNgram(5);
+  encoder.SetminNgram(4);
+  const int val = encoder.MinNgram();
+  encoder.Encode(stringInput, output, tokenizer);
+  const DictionaryType& dictionary = encoder.Dictionary();
+
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t> keysCount;
+  for (auto& keyValue : dictionary.Mapping())
+  {
+    keysCount[keyValue.second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(keysCount[keyValue.second], 1);
+  }
+
```

Comment:
Test verified using sklearn ngram  for range (4,5)

sklearn output

![Screenshot from 2019-11-08 15-19-38](https://user-images.githubusercontent.com/21286134/68468017-385b7580-023d-11ea-84fd-203a6cde9b1d.png)

Mlpack output

![Screenshot from 2019-11-08 15-19-32](https://user-images.githubusercontent.com/21286134/68468027-401b1a00-023d-11ea-9490-ed12feee47ba.png)